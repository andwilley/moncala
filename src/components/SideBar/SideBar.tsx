import React from "react";
import NoteListItem from "../NoteListItem/NoteListItem";

export const SideBar: React.FC = () => {
  return (
    <div className="Sidebar dib w5 h-100 ph2 overflow-y-auto">
      <ul className="NotesList list pl0 mt0">
        {[1, 2, 3, 4].map(e => (
          <NoteListItem />
        ))}
      </ul>
    </div>
  );
};

export default SideBar;
