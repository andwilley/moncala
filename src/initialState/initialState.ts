import { IState } from '../types/IState';
import { EditorState } from 'draft-js';

export const initialState: IState = {
  notes: {
    byId: {},
    allIds: []
  },
  activeId: undefined,
  editorState: EditorState.createEmpty(),
  focusEditor: false
};
