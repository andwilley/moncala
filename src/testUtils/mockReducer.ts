import { IState } from '../types/IState';
import { Action } from '../actions/actions';

const mockReducer = (testState: IState) => (
  state: IState | undefined = testState,
  action: Action
) => state;

export default mockReducer;
