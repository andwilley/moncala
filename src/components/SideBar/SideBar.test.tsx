import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from './SideBar';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import mockReducer from '../../testUtils/mockReducer';
import { initialState } from '../../initialState/initialState';

describe('<SideBar> test a basic render output', () => {
  const store = createStore(mockReducer(initialState));
  it('should render the sidebar', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SideBar />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
