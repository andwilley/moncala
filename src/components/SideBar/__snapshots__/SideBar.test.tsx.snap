// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar> test a basic render output should call dispatch on note click 1`] = `
<div
  className="Sidebar dib w5 h-100 ph2 overflow-y-auto"
>
  <ul
    className="NotesList list pl0 mt0"
  >
    0
  </ul>
</div>
`;

exports[`<SideBar> test a basic render output should render the sidebar 1`] = `
<div
  className="Sidebar dib w5 h-100 ph2 overflow-y-auto"
>
  <ul
    className="NotesList list pl0 mt0"
  />
</div>
`;
